### PR TITLE
feat: add configure function for log level visibility

### PR DESCRIPTION
**Pull request by bgwdotdev**

### CODE REVIEW

**Review by lpil — COMMENTED** [review]
Hello! Could you add use of this function to the test please

**Comment by bgwdotdev** [comment]
No bother! added to the test case and also updated the changelog if that's helpful too ^^;

cheers :)

**Comment by bgwdotdev** [comment]
just as a quick note, I added a little doc for the set_level function in-line with the docs you've added and rebased against main 

Additionally, I was going to update the return type to better return as per erlang docs https://www.erlang.org/doc/man/logger#set_primary_config-2 but I wasn't actually sure how to handle the `term()` type on the error since it's an any type? is there a reference example of how to handle this at all somewhere ^^?

**Comment by lpil** [comment]
Let's use `Dynamic` for `term()` as we don't know what it is. Could be anything, and we don't care about it!

Thank you

**Comment by bgwdotdev** [comment]
ah! of course, forgot about dynamic. Now added that.

Wonder if it would make sense to also add helper functions for log levels to shorthand easier, for example

```gleam
logging.log(logging.Error, "my error")

logging.error("my error")
```

happy to throw in a function for each.

Was also considering if having a formatter for structured logging would make sense? pulling in gleam_json and maybe doing something like

```json

{ "level": "error", "date": 1715380045, "message": "my error" }

```
or something to that effect ^^?

Just some ideas if they were of interest, happy to implement.

File: src/logging.gleam [hunk 1]
```diff
@@ -33,3 +35,19 @@ pub fn log(level: LogLevel, message: String) -> Nil {
...
+
+@external(erlang, "logging_ffi", "set_primary_config")
+fn set_primary_config_level(
+  key: Key,
+  level: LogLevel,
+) -> Result(Nil, dynamic.Dynamic)
+
+/// Change the log visibility level to be output from the default of `Info`.
+///
+pub fn set_level(level: LogLevel) -> Result(Nil, dynamic.Dynamic) {
```
Comment (lpil) [inline src/logging.gleam]: Please revert this to the previous API. I don't want to expose the error value at all, rather silently do nothing in that case like the existing code does.

**Review by lpil — COMMENTED** [review]
> Wonder if it would make sense to also add helper functions for log levels to shorthand easier, for example
>
> Was also considering if having a formatter for structured logging would make sense? pulling in gleam_json and maybe doing something like

This package does nothing but offer some simple defaults for the Erlang logger. There already logging libraries that do these, best to use those instead.

File: src/logging.gleam (hunk 1 above)
Comment (bgwdotdev) [reply src/logging.gleam]: Oh ok! fair enough, will do!

**Comment by bgwdotdev** [comment]
reverted! hopefully good to go now :) thanks for your feedback and time

File: src/logging.gleam [hunk 2]
```diff
@@ -33,3 +33,17 @@ pub fn log(level: LogLevel, message: String) -> Nil {
 
 @external(erlang, "logger", "log")
 fn erlang_log(level: LogLevel, message: String) -> DoNotLeak
+
+@external(erlang, "logger", "set_primary_config")
+fn set_primary_config_level(key: Key, level: LogLevel) -> Nil
+
+/// Change the log visibility level to be output from the default of `Info`.
```
Comment (lpil) [inline src/logging.gleam]: `Info` isn't the default for the Erlang logger, so let's remove that bit.

File: src/logging.gleam [hunk 3]
```diff
@@ -33,3 +33,17 @@ pub fn log(level: LogLevel, message: String) -> Nil {
 
 @external(erlang, "logger", "log")
 fn erlang_log(level: LogLevel, message: String) -> DoNotLeak
+
+@external(erlang, "logger", "set_primary_config")
+fn set_primary_config_level(key: Key, level: LogLevel) -> Nil
```
Comment (lpil) [inline src/logging.gleam]:
```suggestion
fn set_primary_config_level(key: Key, level: LogLevel) -> DoNotLeak
```

**Review by lpil — APPROVED** [review]
Thank you 💜

File: src/logging.gleam (hunk 3 above)
Comment (bgwdotdev) [reply src/logging.gleam]: I'm curious on this bit, is this just a type purely set for developer documentation purposes? not doing any special stuff in the ffi or anything?

**Review by lpil — APPROVED** [review]
Thank you!!

File: src/logging.gleam (hunk 3 above)
Comment (lpil) [reply src/logging.gleam]: It's a private type defined above to ensure that any unknown value doesn't leak in to the public interface. The previous typing was incorrect and due to being public could leak due to a programmer mistake, which could potentially cause runtime issues.

**Pull request merged**